diff.py

```diff
@@ -1114,9 +1123,22 @@ def run_make_capture_output(
 
 
 def restrict_to_function(dump: str, fn_name: str) -> str:
+
+    # Regex objdump's output to find the beginning of the line which contains the offset and symbol
+    # "OFFSET <SYM>:"
+    restrictFunctionRegex = f"[0-9A-Fa-f]+ <{fn_name}>:"
```

Comment:
fn_name needs escaping to be part of a regex, and also searching for a regex seems a bit unnecessarily slow. Can we keep searching as before but then search backwards for a newline?